Competition2018/src/main/java/competition/subsystems/offboard/OffboardCommsCanInterface.java

```diff
@@ -0,0 +1,55 @@
+package competition.subsystems.offboard;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.IntBuffer;
+
+import org.apache.log4j.Logger;
+
+import edu.wpi.first.wpilibj.can.CANInvalidBufferException;
+import edu.wpi.first.wpilibj.can.CANJNI;
+import edu.wpi.first.wpilibj.can.CANMessageNotAllowedException;
+import edu.wpi.first.wpilibj.can.CANMessageNotFoundException;
+import edu.wpi.first.wpilibj.can.CANNotInitializedException;
+
+public class OffboardCommsCanInterface implements XOffboardCommsInterface {
```

Comment:
It may be too late now, but in general I'd recommend having XOffboardCommsInterface be an abstract class instead with all of this logic into it, with the exception of something like

`abstract void SendMessageViaCAN(arbitrationID, data, period)`
`abstract void GetMessageViaCAN(buffer, mask, time)`

where that's the only method implemented by the concrete class. That makes all the logic you wrote in this class easily unit-testable.